src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -528,6 +529,33 @@ def migrate_alert_rule(
     )
 
 
+def get_workflow_name(alert_rule: AlertRule) -> str:
+    # max length of name is 256 characters
+    # get triggers -> get actions
+    # this is a placeholder replicating the existing behavior; changes to the workflow
+    # name will come in a subsequent PR
+    return alert_rule.name
+
+
+def dual_write_alert_rule(alert_rule: AlertRule, user: RpcUser | None = None) -> None:
+    """
+    Comprehensively dual write the ACI objects corresponding to an alert rule, its triggers, and
+    its actions. All these objects will have been created before calling this method.
+    """
+    # step 1: migrate the alert rule
+    migrate_alert_rule(alert_rule)
+    triggers = AlertRuleTrigger.objects.filter(alert_rule=alert_rule)
+    # step 2: migrate each trigger
+    for trigger in triggers:
+        migrate_metric_data_conditions(trigger)
+        trigger_actions = AlertRuleTriggerAction.objects.filter(alert_rule_trigger=trigger)
+        # step 3: for each trigger, migrate the actions
+        for trigger_action in trigger_actions:
+            migrate_metric_action(trigger_action)
+    # step 4: migrate alert rule resolution
+    migrate_resolve_threshold_data_conditions(alert_rule)
```

Comment:
nit: we might want to wrap this in a transction - so we can ensure all the `migrate` things are successful.